ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto

```diff
@@ -128,6 +135,46 @@ message GetFlatTransactionResponse {
   Transaction transaction = 1;
 }
 
+message GetContractByIdRequest {
+  // The contract id being queried
+  // Required
+  string contract_id = 1;
+}
+message GetContractByIdResponse {
+  // The collection of events, will only contain consumed ExerciseEvents
+  repeated TreeEvent events = 1;
+}
+
+message GetContractByKeyRequest {
+
+  // The contract key to search for.  Note that only contracts that are visible to the
+  // requesting parties will returned.  To ensure that no contract key assignments are missed
+  // request as a key maintainer.
+  // Required
+  Value contract_key = 1;
```

Comment:
I think you are going to need template id too, contract key value does not include it

Reply:
The database lookup does not require template id and one looked up the template id is available in event tables.

Reply:
The DB table does not require it, but you'll require a `templateId` for computing the hash to lookup: https://github.com/digital-asset/daml/blob/440b582ebe4d524a316b64aaac0017f67a41b11a/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala#L310-L314

Reply:
Done.